src/ORM/FieldType/DBCurrency.php

```diff
@@ -53,15 +59,14 @@ public function Whole(): string
 
     public function setValue(mixed $value, null|array|ModelData $record = null, bool $markChanged = true): static
     {
-        $matches = null;
-        if (is_numeric($value)) {
-            $this->value = $value;
-        } elseif (preg_match('/-?\$?[0-9,]+(.[0-9]+)?([Ee][0-9]+)?/', $value ?? '', $matches)) {
-            $this->value = str_replace(['$', ',', static::config()->get('currency_symbol')], '', $matches[0] ?? '');
-        } else {
-            $this->value = 0;
+        parent::setValue($value, $record, $markChanged);
+        if (is_string($this->value)) {
+            $symbol = static::config()->get('currency_symbol');
+            $val = str_replace(['$', ',', $symbol], '', $this->value);
+            if (is_numeric($val)) {
+                $this->value = (float) $val;
+            }
         }
```

Comment:
```suggestion
        if (is_string($this->value)) {
            $symbol = static::config()->get('currency_symbol');
            $val = str_replace(['$', ',', $symbol], '', $this->value);
            if (is_numeric($val)) {
                $value = (float) $val;
            }
        }
        parent::setValue($value, $record, $markChanged);
```
Don't set value twice, and _do_ set the value if it's not "valid" rather than just not setting anything, or else validation will never get to validate it.

Reply:
Updated